benchcab/utils/pbs.py

```diff
@@ -13,35 +11,30 @@ def render_job_script(
     config_path: str,
     modules: list,
     benchcab_path: str,
+    pbs_config: dict,
     verbose=False,
     skip_bitwise_cmp=False,
-    pbs_config: Optional[dict] = None,
 ) -> str:
     """Returns the text for a PBS job script that executes all computationally expensive commands.
 
     This includes things such as running CABLE and running bitwise comparison jobs
     between model output files.
     """
-    if pbs_config is None:
-        pbs_config = internal.FLUXSITE_DEFAULT_PBS
+    pbs_missing_keys = internal.FLUXSITE_DEFAULT_PBS.keys() - pbs_config.keys()
+    if len(pbs_missing_keys) != 0:
+        msg = f"Default pbs parameters missing: {sorted(pbs_missing_keys)}"
+        raise ValueError(msg)
```

Comment:
Sorry, I missed this one at the first go. It's not what we want to happen. If people give only one parameter, then the other ones should be set to the default values. So I guess it means the `read_optional_data` function needs to do that.

Reply:
Yes, `read_optional_data` does that (The `|` operator takes the union of all keys of LHS and RHS, but if RHS and LHS have same key, the RHS value is chosen)

Reply:
So I'll add unit tests in `test_config.py` which checks that the different optional parameters have been added

Reply:
OK that's fine.